http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -155,19 +155,35 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
     val resourceTypes = resources.map(r => sr.resourceType(r._2)).toList.toSet
     for {
-      projectPolicies <- samDao.getResourcePolicies[ProjectSamResourceId](authHeader, SamResourceType.Project)
-      owningProjects = projectPolicies.collect { case (r, SamPolicyName.Owner) =>
+      projectPolicies <- samDao
+        .getResourcePolicies[ProjectSamResourceId](authHeader, SamResourceType.Project)
+      readingProjects = projectPolicies.map(_._1.googleProject)
+      _ <- logger.info(s"Reading: $readingProjects")
+      ownedProjects = projectPolicies.collect { case (r, SamPolicyName.Owner) =>
         r.googleProject
       }
+      _ <- logger.info(s"Owned: $ownedProjects")
       resourcePolicies <- resourceTypes.toList.flatTraverse(resourceType =>
         samDao.getResourcePolicies[R](authHeader, resourceType)
       )
+      _ <- logger.info(s"resourcePolicies: $resourcePolicies")
       res = resourcePolicies.filter { case (r, pn) => sr.policyNames(r).contains(pn) }
     } yield resources.filter { case (project, r) =>
-      owningProjects.contains(project) || res.exists(_._1 == r)
+      // user must be reader on project and resource, or a project owner
```

Comment:
Is this check redundant with the `isUserProjectReader below`?

Reply:
I don't know if I understand the redundancy, this is filtering out resources where the user isn't a reader of the associated project